pkg/cmd/infra/router/router.go

```diff
@@ -27,28 +28,104 @@ This command launches a router connected to your cluster master. The router list
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
-type templateRouterConfig struct {
-	Config             *clientcmd.Config
-	TemplateFile       string
-	ReloadScript       string
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
+	// Config holds global flags that are passed to all client commands that need
+	// to connect to master.
+	Config *clientcmd.Config
+
+	// Type is the type of plugin being created: f5 or template.
+	Type string
```

Comment:
I understand that part, but the bindings would be to some `cfg.TemplateRouterConfig.foo` or some `cfg.F5RouterConfig.foo`, and we don't know whether to create `cfg.TemplateRouterConfig` or `cfg.F5RouterConfig` until we have read the type flag, which happens after flags are bound, right?